packages/backend/src/helpers/compute-parameters.ts

```diff
@@ -55,11 +56,25 @@ function findAndSubstituteVariables(
       if (isVariable) {
         const stepIdAndKeyPath = part.replace(/{{step.|}}/g, '') as string
         const [stepId, ...keyPaths] = stepIdAndKeyPath.split('.')
-        const keyPath = keyPaths.join('.')
         const executionStep = executionSteps.find((executionStep) => {
           return executionStep.stepId === stepId
         })
         const data = executionStep?.dataOut
+
+        // custom logic to deal with key encoding for data from vault
+        if (
+          executionStep.appKey === vaultWorkspace.key &&
+          data &&
+          data._metadata &&
+          (data._metadata as Record<string, any>).keysEncoded
+        ) {
+          keyPaths[keyPaths.length - 1] = Buffer.from(
```

Comment:
Wait, won't this break newly created pipes? Because new pipes will refer to the hex-encoded column name in `dataOut`, and not the decoded column name.